src/core/image.js

```diff
@@ -79,6 +79,68 @@ var PDFImage = (function PDFImageClosure() {
     return dest;
   }
 
+  /**
+   * Downsizes an image mask.
+   * @param {TypedArray} src - The source buffer.
+   * @param {Number} w1 - Original width.
+   * @param {Number} h1 - Original height.
+   * @param {Number} w2 - New width.
+   * @param {Number} h2 - New height.
+   * @returns {Object} Downsized image mask buffer.
+   */
+  function downsizeImageMask(src, w1, h1, w2, h2) {
+    let resizeFactorX = Math.round(w1 / w2);
+    let resizeFactorY = Math.round(h1 / h2);
+    if (resizeFactorX === 1 && resizeFactorY === 1) {
+      return null;
+    }
+    let packedWidth = (w1 + 7) >> 3;
+    let xHop = (resizeFactorX >> 3);
+    xHop = xHop ? xHop : 1;
```

Comment:
Can't you simply fold the two lines above together as `let xHop = (resizeFactorX >> 3) || 1;`, rather than first initializing it and then using a ternary condition?